From: Chris Renhsaw <osm0sis@outlook.com>
Date: Tue, 11 Apr 2017 07:50:46 -0300
Subject: [PATCH] hack: use dietlibc's resolver to avoid static compiliation
 issues with bionic's getaddrinfo() returning only null instead of opening the
 expected connection to /dev/socket/dnyproxyd

---
 libbb/xconnect.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/libbb/xconnect.c b/libbb/xconnect.c
index 3a0dc26..52e5e19 100644
--- a/libbb/xconnect.c
+++ b/libbb/xconnect.c
@@ -12,6 +12,8 @@
 #include <netinet/in.h>
 #include <net/if.h>
 #include <sys/un.h>
+#include <netdb.h>
+#include "../libres/dietdns.h"
 #include "libbb.h"
 
 int FAST_FUNC setsockopt_int(int fd, int level, int optname, int optval)
@@ -282,7 +284,7 @@ IF_NOT_FEATURE_IPV6(sa_family_t af = AF_INET;)
 	 * for each possible socket type (tcp,udp,raw...): */
 	hint.ai_socktype = SOCK_STREAM;
 	hint.ai_flags = ai_flags & ~DIE_ON_ERROR;
-	rc = getaddrinfo(host, NULL, &hint, &result);
+	rc = diet_getaddrinfo(host, NULL, &hint, &result); /* hack to use dietlibc's resolver to work around static compile issue */
 	if (rc || !result) {
 		bb_error_msg("bad address '%s'", org_host);
 		if (ai_flags & DIE_ON_ERROR)
@@ -309,7 +311,7 @@ IF_NOT_FEATURE_IPV6(sa_family_t af = AF_INET;)
 	set_nport(&r->u.sa, htons(port));
  ret:
 	if (result)
-		freeaddrinfo(result);
+		diet_freeaddrinfo(result); /* hack to use dietlibc's resolver to work around static compile issue */
 	return r;
 }
 #if !ENABLE_FEATURE_IPV6
-- 
2.8.3

